src/common/file.cpp

```diff
@@ -291,6 +300,14 @@ namespace tools {
       return true;
     }
 
+    fs::path depreciated_path = get_depreciated_default_data_dir();
+    if (fs::is_directory(depreciated_path, ec))
+    {
+      fs::rename(depreciated_path, path);
+      fs::create_directory_symlink(depreciated_path, path);
```

Comment:
This looks reversed: the first argument is the target (i.e. what the symlink points to), the second is the symlink filename itself.

However, I'm concerned that this `create_directories_if_necessary` is going to be called enen when the user gave an explicit `--data-dir` option, so we want to update this method to take a `bool is_default` and only do the rename+symlink if `is_default`.

Also: `create_directory_symlink` is almost certainly going to fail on Windows, so `#ifndef WIN32` around the symlink call.